### ISSUE TITLE
Not able to run anbox on ubuntu 20.04

### ISSUE THREAD

**Issue by Aditya1927** [anbox/anbox#1735]
sudo add-apt-repository ppa:morphis/anbox-support
 
 More info: https://launchpad.net/~morphis/+archive/ubuntu/anbox-support
Press [ENTER] to continue or Ctrl-c to cancel adding it.

Hit:1 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:2 http://archive.ubuntu.com/ubuntu focal InRelease                         
Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease                 
Ign:4 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal InRelease
Hit:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease         
Hit:6 http://archive.ubuntu.com/ubuntu focal-security InRelease          
Err:7 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release
  404  Not Found [IP: 91.189.95.85 80]
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

**Comment by jnnksdev** [comment]
The apt package is outdated unfortunately.
You can use the snap package or build it from source.

**Comment by calidion** [comment]
@jnnksdev snap is not running either.

**Comment by nicolalandro** [comment]
I build [anbox-modules] and seams it works. I also build anbox. but it does not work the error is on the container:
```
$ sudo anbox container-manager                                       10:10:02
[sudo] password di xxx:          
[ 2021-01-29 09:10:13] [container_manager.cpp:74@operator()] You are running the container manager manually which is most likely not
[ 2021-01-29 09:10:13] [container_manager.cpp:75@operator()] what you want. The container manager is normally started by systemd or
[ 2021-01-29 09:10:13] [container_manager.cpp:76@operator()] another init system. If you still want to run the container-manager
[ 2021-01-29 09:10:13] [container_manager.cpp:77@operator()] you can get rid of this warning by starting with the --daemon option.
[ 2021-01-29 09:10:13] [container_manager.cpp:78@operator()] 
[ 2021-01-29 09:10:13] [container_manager.cpp:122@operator()] boost::filesystem::create_directories: Invalid argument
```

How can I solve this issue? (Maybe I think that the difference can be the kernel, I have the latest version on apt)

**Comment by Nishant0073** [comment]
> sudo add-apt-repository ppa:morphis/anbox-support
> 
> More info: https://launchpad.net/~morphis/+archive/ubuntu/anbox-support
> Press [ENTER] to continue or Ctrl-c to cancel adding it.
> 
> Hit:1 http://dl.google.com/linux/chrome/deb stable InRelease
> Hit:2 http://archive.ubuntu.com/ubuntu focal InRelease
> Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
> Ign:4 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal InRelease
> Hit:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease
> Hit:6 http://archive.ubuntu.com/ubuntu focal-security InRelease
> Err:7 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release
> 404 Not Found
> Reading package lists... Done
> E: The repository 'http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release' does not have a Release file.
> N: Updating from such a repository can't be done securely, and is therefore disabled by default.
> N: See apt-secure(8) manpage for repository creation and user configuration details.

Yes,I had faced same problem add repo  not working but you can do it by another way 
Just do following things:

```
git clone https://github.com/anbox/anbox-modules
cd anbox-modules
sudo chmod +x INSTALL.sh
./INSTALL.sh
```

**Comment by nicolalandro** [comment]
Yes I do exactly that, after I try with repo version of anbox or with my compile version of anbox and I have the same Issue:

```
$ anbox session-manager                        
[ 2021-01-29 10:23:35] [daemon.cpp:61@Run] Failed to connect to socket /run/anbox-container.socket: No such file or directory
```
So I try to run the container manually:
```
$ sudo anbox container-manager       
[sudo] password di xxx:          
[ 2021-01-29 09:10:13] [container_manager.cpp:74@operator()] You are running the container manager manually which is most likely not
[ 2021-01-29 09:10:13] [container_manager.cpp:75@operator()] what you want. The container manager is normally started by systemd or
[ 2021-01-29 09:10:13] [container_manager.cpp:76@operator()] another init system. If you still want to run the container-manager
[ 2021-01-29 09:10:13] [container_manager.cpp:77@operator()] you can get rid of this warning by starting with the --daemon option.
[ 2021-01-29 09:10:13] [container_manager.cpp:78@operator()] 
[ 2021-01-29 09:10:13] [container_manager.cpp:122@operator()] boost::filesystem::create_directories: Invalid argument
```
So The first command does not work because the second does not? Have you any idea?

**Comment by Nishant0073** [comment]
> Failed to connect to socket /run/anbox-container.socket: No such file or directory

Go to this issue [](https://github.com/anbox/anbox/issues/123)

**Comment by sajjad606** [comment]
hi, if you are in iran you can pay any forign payments via our web site.
سایت EPAY 724 | خدمات ارزی بین الملل (https://bit.ly/3uieWwW) (سایت EPAY 724 | خدمات ارزی بین الملل (https://epay724.com))
online payments for application fees, buying gift cards, international exams fees are easier than ever ….

**Comment by Nishant0073** [comment]
> Yes I do exactly that, after I try with repo version of anbox or with my compile version of anbox and I have the same Issue:
> 
> ```
> $ anbox session-manager                        
> [ 2021-01-29 10:23:35] [daemon.cpp:61@Run] Failed to connect to socket /run/anbox-container.socket: No such file or directory
> ```
> 
> So I try to run the container manually:
> 
> ```
> $ sudo anbox container-manager       
> [sudo] password di xxx:          
> [ 2021-01-29 09:10:13] [container_manager.cpp:74@operator()] You are running the container manager manually which is most likely not
> [ 2021-01-29 09:10:13] [container_manager.cpp:75@operator()] what you want. The container manager is normally started by systemd or
> [ 2021-01-29 09:10:13] [container_manager.cpp:76@operator()] another init system. If you still want to run the container-manager
> [ 2021-01-29 09:10:13] [container_manager.cpp:77@operator()] you can get rid of this warning by starting with the --daemon option.
> [ 2021-01-29 09:10:13] [container_manager.cpp:78@operator()] 
> [ 2021-01-29 09:10:13] [container_manager.cpp:122@operator()] boost::filesystem::create_directories: Invalid argument
> ```
> 
> So The first command does not work because the second does not? Have you any idea?
1st dowlnoad the anbox-modules & install it.
2nd install anbox beta version
and then run
anbox session-manager
anbox.appmgr

**Comment by Nishant0073** [comment]
> Yes I do exactly that, after I try with repo version of anbox or with my compile version of anbox and I have the same Issue:
> 
> ```
> $ anbox session-manager                        
> [ 2021-01-29 10:23:35] [daemon.cpp:61@Run] Failed to connect to socket /run/anbox-container.socket: No such file or directory
> ```
> 
> So I try to run the container manually:
> 
> ```
> $ sudo anbox container-manager       
> [sudo] password di xxx:          
> [ 2021-01-29 09:10:13] [container_manager.cpp:74@operator()] You are running the container manager manually which is most likely not
> [ 2021-01-29 09:10:13] [container_manager.cpp:75@operator()] what you want. The container manager is normally started by systemd or
> [ 2021-01-29 09:10:13] [container_manager.cpp:76@operator()] another init system. If you still want to run the container-manager
> [ 2021-01-29 09:10:13] [container_manager.cpp:77@operator()] you can get rid of this warning by starting with the --daemon option.
> [ 2021-01-29 09:10:13] [container_manager.cpp:78@operator()] 
> [ 2021-01-29 09:10:13] [container_manager.cpp:122@operator()] boost::filesystem::create_directories: Invalid argument
> ```
> 
> So The first command does not work because the second does not? Have you any idea?


1st dowlnoad the anbox-modules & install it.
2nd install anbox beta version
and then run
anbox session-manager
anbox.appmgr

**Comment by connected201** [comment]
> 
> 
> > sudo add-apt-repository ppa:morphis/anbox-support
> > More info: https://launchpad.net/~morphis/+archive/ubuntu/anbox-support
> > Press [ENTER] to continue or Ctrl-c to cancel adding it.
> > Hit:1 http://dl.google.com/linux/chrome/deb stable InRelease
> > Hit:2 http://archive.ubuntu.com/ubuntu focal InRelease
> > Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
> > Ign:4 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal InRelease
> > Hit:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease
> > Hit:6 http://archive.ubuntu.com/ubuntu focal-security InRelease
> > Err:7 http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release
> > 404 Not Found
> > Reading package lists... Done
> > E: The repository 'http://ppa.launchpad.net/morphis/anbox-support/ubuntu focal Release' does not have a Release file.
> > N: Updating from such a repository can't be done securely, and is therefore disabled by default.
> > N: See apt-secure(8) manpage for repository creation and user configuration details.
> 
> Yes,I had faced same problem add repo not working but you can do it by another way
> Just do following things:
> 
> ```
> git clone https://github.com/anbox/anbox-modules
> cd anbox-modules
> sudo chmod +x INSTALL.sh
> ./INSTALL.sh
> ```
> 
> Then run
> ` anbox session-manager`
> 
> keep this terminal running and in another tab run
> 
> `anbox.appmgr`
> 
> I wish this will resolve your problem 👍

Hi, how about ls -1 /dev/{ashmem,binder} ?
how you can install it?
because i have error
![image](https://user-images.githubusercontent.com/20506433/111193797-3c32a100-85c3-11eb-8a60-084f8500cddc.png)
and to resolve this error i need to make
$ sudo add-apt-repository ppa:morphis/anbox-support
$ sudo apt update
$ sudo apt install linux-headers-generic anbox-modules-dkms

**Comment by warrenc5** [comment]
something didn't compile correctly - can you grep the make.log for error? Maybe don't post the entire file - (It is too big)

**Comment by Rubix-1** [comment]
can someone see the exact fix for this I cant seem to find another way :(

**Comment by tommytran732** [comment]
The snap package worked fine for me, though ashmem_linux is not signed by default for whatever reason, even when it is shipped with Ubuntu out of the box. These commands should fix it:

`sudo kmodsign sha512 /var/lib/shim-signed/mok/MOK.priv /var/lib/shim-signed/mok/MOK.der /lib/modules/`uname -r`/kernel/drivers/staging/android/ashmem_linux.ko`

`sudo modprobe ashmem_linux`
